.github/workflows/ci.yml

```diff
@@ -32,73 +32,191 @@ jobs:
 
   injection-tests-java:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     strategy:
       matrix:
-        lib-injection-connection: ['uds', 'network']
+        lib-injection-connection: [ 'network','uds']
         lib-injection-use-admission-controller: ['', 'use-admission-controller']
+        weblog-variant: [ 'dd-lib-java-init-test-app']
       fail-fast: false
     env:
       TEST_LIBRARY: java
+      WEBLOG_VARIANT: ${{ matrix.weblog-variant }}
       LIBRARY_INJECTION_CONNECTION: ${{ matrix.lib-injection-connection }}
-      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-admission-controller }}
-      LIBRARY_INJECTION_INIT_IMAGE: ghcr.io/datadog/dd-trace-java/dd-lib-java-init:34db2f76715dbbe9168806bac7c89c7ad4d51996
-      LIBRARY_INJECTION_TEST_APP_IMAGE: ghcr.io/datadog/dd-trace-java/dd-lib-java-init-test-app:34db2f76715dbbe9168806bac7c89c7ad4d51996
+      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-use-admission-controller }}
+      DOCKER_REGISTRY_IMAGES_PATH: ghcr.io/datadog/system-tests
```

Comment:
Not sure it makes sense to abstract this, the init image and the test app don't necessarily have to be published to the same registry.